helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java

```diff
@@ -148,6 +133,8 @@ protected static RealmAwareZkClient createZkClient(HelixManager manager) {
     }
     RealmAwareZkClient.RealmAwareZkClientConfig clientConfig =
         new RealmAwareZkClient.RealmAwareZkClientConfig().setZkSerializer(new ZNRecordSerializer());
+    // Set operation retry timeout to prevent hanging infinitely
+    clientConfig.setOperationRetryTimeout((long) 60000);
```

Comment:
Question: Are 60 secs good enough? There could transient zk connection issues. My suggestion is make it a bit longer, eg. 5, 10, or 30 mins, to have a higher rate of success.
And a code style suggestion, try to use java `Duration` instead of this hard coded number, which will make it much easier to read/maintain.
```
Duration.ofMinutes(10).toMillis()
```

The lazy initialization way mentioned by @jiajunwang was also what I was thinking of when I debugged the code here. Too many things are done for constructing an object was my first impression: creating a ZkClient, connecting to zk and reading the znode. If I did, I'd prefer to lazily construct the threadpool/reading znode after this factory object is constructed. Can be done later.

Reply:
That's a good suggestion on code style, thanks. I will go with 5 minutes.

Lazy construction was considered when this feature was first developed. It requires a larger scale of code change, so it won't be done in this bugfix PR. It can be done later, though, like you said.